### PR TITLE
Investments enhancements

### PR DESCRIPTION
**Pull request by Crashillo**
## :v: What does this PR do?
Includes some improvements for investments module:
- Add a new fieldtype for the configuration object called `separator`. Closes https://github.com/PopulateTools/issues/issues/824
- Remove some attrs from configuration, and filters non-null values. Closes https://github.com/PopulateTools/issues/issues/823
- Metatitle fixed. Closes https://github.com/PopulateTools/issues/issues/822
- Show `estat` in gallery. Also filters non-null values. Closes https://github.com/PopulateTools/issues/issues/821
- Keep user selection on filters when back and forth from home to project. Closes https://github.com/PopulateTools/issues/issues/819
- Update the number of items by checkbox element, after modifying the filters. Closes https://github.com/PopulateTools/issues/issues/835
- Reset filters to initial values (empty). Closes https://github.com/PopulateTools/issues/issues/834
- Enables a button to expand/collapse all items list. Closes https://github.com/PopulateTools/issues/issues/833
- Show a blurred background when there's only one item in the gallery. It's only apreciated when the image is vertical. Closes https://github.com/PopulateTools/issues/issues/818
- Allow formatting for subfields. Closes https://github.com/PopulateTools/issues/issues/836

### 🎈 Caveats
- Pending PRs for this module were merged, so some issues has been already fixed
- Documentation may be found [here](https://gobierto.readme.io/docs/inversiones-configuration)

### CODE REVIEW

File: app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  ],
+  itemSpecialConfiguration: {
+    fn: async data => {
+      const dataArr = Array.isArray(data) ? data : [data];
+      const filterArr = dataArr.filter(d => d.attributes.partida && d.attributes["any-partida"]);
+      const allIds = filterArr.map(d => `'${d.attributes.partida}'`).join(",");
+
+      // Avoid the query if there's no code_with_zone (partida)
+      if (allIds) {
+        const endpoint = `/api/v1/data`;
```
Comment (ferblape) [inline app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js]: This endpoint is broken now, right? New url is `/api/v1/data/data`?

File: app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
+import axios from "axios";
+
+export default {
```
Comment (ferblape) [inline app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js:3]: Is there an issue to extract this custom JS to somewhere outside the base code?

Comment (Crashillo) [reply app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js:3]: Nope. Only you must update properly in the `mixin/common.js` file the path where is located this configuration.

But, this file is a technical debt at the moment. It shouldn't exist, it currently plays the middleware role, but such configuration must be provided from backoffice, and the property `itemSpecialConfiguration` must be included in the data endpoint itself. These changes require ETL, API and front-end modifications.
There was an issue regarding this, but it's outdated: https://github.com/PopulateTools/issues/issues/796 (I'm gonna update it with these points)

Comment (Crashillo) [reply app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js:3]: Regarding your question, yes, we can place it somewhere else, but does it worth? Do we proceed? If so, where do we place it?

File: app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js (hunk 1 above)
Comment (Crashillo) [reply app/javascript/gobierto_investments/webapp/mixins/mataro.conf.js]: Done

**Review by entantoencuanto — APPROVED** [review]

**Pull request merged**